lib/srv/uacc/uacc.h

```diff
@@ -59,6 +61,30 @@ static char* get_absolute_path_with_fallback(char* buffer, const char* supplied_
     return realpath(path, buffer);
 }
 
+static int check_abs_path_err(const char* buffer) {
+    // check for errors
+    if (buffer == NULL) {
+        switch errno {
+            case EACCES: return UACC_UTMP_MISSING_PERMISSIONS;
+            case EINVAL: return UACC_UTMP_FAILED_OPEN;
+            case EIO: return UACC_UTMP_READ_ERROR;
+            case ELOOP: return UACC_UTMP_FAILED_OPEN;
+            case ENAMETOOLONG: return UACC_UTMP_FAILED_OPEN;
+            case ENOENT: return UACC_UTMP_PATH_DOES_NOT_EXIST;
+            case ENOTDIR: return UACC_UTMP_FAILED_OPEN;
+            default: return UACC_UTMP_OTHER_ERROR;
+        }
+    }
+
+    // check for GNU extension errors
+    if (errno == EACCES || errno == ENOENT) {
+        return UACC_UTMP_OTHER_ERROR;
+    }
```

Comment:
why is `errno` checked differently based on `buffer`?

Reply:
It's checked differently because a non `NULL` buffer and a non-zero `errno` indicates a GNU extension in glibc error as per the manpage.